[DNM] closedts: shorten target_duration to 5s

ajwerner opened the pull request:
I want to see what tests fail, if any.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/34931)
<!-- Reviewable:end -->

danhhz commented:
Hopefully no unit tests fail with this, it would mean they were taking more than 5s, no? You may want to run some roachtests with this too, but I'm not the expert on which ones

ajwerner commented:
yep, now to launch a battery of roachtests and see what falls out

ajwerner commented:
This is on hold and won't get in for 19.1 due to concerns about long-running schema changes.

Commit:
closedts: shorten target_duration from 30s to 3s

This PR dramatically shortens the closed timestamp target interval. With this
setting the experimental_follower_read_timestamp() will now be 4.8 seconds in
the past as opposed to the current 48 seconds in the past. This relatively
aggressive setting of 3s is intended to shake out issues.

This value has been verified to work on the `schemachange/mixed/tpcc` roachtest
introduced in #39096 and for vanilla TPC-C runs around where we have
established a baseline in the past using tpccbench. Specifically several runs
at 2300 warehouses on 3x c5d.4xlarge nodes which is right at the passing
boundary without the change were run only a very small difference in
efficiency or tail latency was observed.

It seems reasonable to attempt to live with this value on master for a while
and see what happens.

Fixes #37083.

Release note (enterprise change): Shorten the default closed timestamp target
interval from 30s to 3s leading to permit follower reads at 4.8 seconds in
the past rather than the previous 48 seconds.
